Base/QTApp/qSlicerApplicationHelper.txx

```diff
@@ -93,6 +93,17 @@ void qSlicerApplicationHelper::postInitializeApplication(
   enableMainWindow = enableMainWindow && !app.commandOptions()->runPythonAndExit();
   bool showSplashScreen = !app.commandOptions()->noSplash() && enableMainWindow;
 
+
+  if (enableMainWindow && !app.testAttribute(qSlicerCoreApplication::AA_EnableTesting))
```

Comment:
should we check independently of the main window ?

Reply:
I would not want to show a popup when somebody just uses Slicer command-line, potentially not rendering anything. It could make sense to log a warning though.

Reply:
Gotcha. I was thinking of the case of application no having a main window but instantiating their own UI using python. That said this is still experimental and a corner case (I have  a topic I need to finalize ....)

Reply:
qSlicerApplicationHelper::checkRenderingCapabilities() is a public static method that slicelets can call.

Reply:
Indeed, we should be able to add a decorator for the method. The assumption is that no VTK initialization will be done before the python code is executed. 

All of  that said, there are way to address this and it should not block integration of this PR.